core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -355,6 +355,13 @@ private[spark] object SerDe {
           writeInt(dos, v.length)
           v.foreach(elem => writeObject(dos, elem))
 
+        // Handle Properties
```

Comment:
As explained above, this is needed to properly handle java.util.Properties. This is useful for 2 parts:
1. R code could set or get values from java.util.Properties directly
2. For callJMethod to match parameter types properly

As above, we could have a Scala helper that takes in all the parameters - in such case it would be better to have all the logic in Scala and perhaps it would be easier to test.